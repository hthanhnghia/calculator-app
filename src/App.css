.calculatorContainer {
  display: grid;
  width: 300px;
  height: 300px;
  margin: auto;
  grid-template-columns: auto auto auto auto;
  grid-template-rows: 20% 20% 20% 20% 20%;
  border-left: 1px solid red;
  border-bottom: 1px solid red;
}

.calculatorContainer > .grid-item {
  border-top: 1px solid red;
  border-right: 1px solid red;
  display: flex;
  align-items: center;
  justify-content: center;
}

.operators > .grid-item {
  border-right: 1px solid red;
  border-top: 1px solid red;
  display: flex;
  align-items: center;
  justify-content: center;
}

.calculatorContainer > .result {
  grid-column: 1 / span 4;
  grid-row: 1;
  background-color: brown;
  color: white;
  justify-content: flex-end;
}

.clear-calculator {
  grid-column: 1 / span 3;
  grid-row: 2;
}

.numbers {
  grid-column: 1 / span 3;
  grid-row: 3 / span 3;
}

.operators {
  grid-column: 4;
  grid-row: 2 / span 4;
  display: grid;
  grid-template-rows: 25% 25% 25% 25%;
}

.number {
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}