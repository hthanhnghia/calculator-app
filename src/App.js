import React, { Component } from 'react';
import Number from './Number'
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      result: 0
    }
  }

  handleClick = (number) => {
    this.setState({result: number})
  }

  createNumberDiv = () => {
      let numbers = []
      for (let i=1; i<=9; i++) {
        numbers.push(
          <div className="grid-item" key={i}>
            <Number value={i} onClick={()=>this.handleClick(i)} />
          </div>
        )
      }
      return numbers
  }

  render() {
    return (
      <div className="calculatorContainer">
        <div className="grid-item result">
          {this.state.result}
        </div>

        <div className="grid-item clear-calculator">
          clear
        </div>

        {this.createNumberDiv()}

        <div className="operators">
          <div className="grid-item">:</div>
          <div className="grid-item">-</div>
          <div className="grid-item">+</div>
          <div className="grid-item">=</div>
        </div>
      </div>
    );
  }
}

export default App;
