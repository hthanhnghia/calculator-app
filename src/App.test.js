import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders a snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Number component render number 0', () => {
  const tree = renderer.create(<Number value="0" />).toJSON();
  expect(tree).toMatchSnapshot();
})