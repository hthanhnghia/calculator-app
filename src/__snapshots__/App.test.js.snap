// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number component render number 0 1`] = `"NaN"`;

exports[`renders a snapshot 1`] = `
<div
  className="container"
  onKeyDown={[Function]}
  tabIndex="0"
>
  <div
    className="calculatorContainer"
  >
    <div
      className="grid-item result"
    >
      0
    </div>
    <div
      className="grid-item clear-calculator"
      onClick={[Function]}
    >
      clear
    </div>
    <div
      className="grid-item"
    >
      <div
        className="number"
        onClick={[Function]}
      >
        1
      </div>
    </div>
    <div
      className="grid-item"
    >
      <div
        className="number"
        onClick={[Function]}
      >
        2
      </div>
    </div>
    <div
      className="grid-item"
    >
      <div
        className="number"
        onClick={[Function]}
      >
        3
      </div>
    </div>
    <div
      className="grid-item"
    >
      <div
        className="number"
        onClick={[Function]}
      >
        4
      </div>
    </div>
    <div
      className="grid-item"
    >
      <div
        className="number"
        onClick={[Function]}
      >
        5
      </div>
    </div>
    <div
      className="grid-item"
    >
      <div
        className="number"
        onClick={[Function]}
      >
        6
      </div>
    </div>
    <div
      className="grid-item"
    >
      <div
        className="number"
        onClick={[Function]}
      >
        7
      </div>
    </div>
    <div
      className="grid-item"
    >
      <div
        className="number"
        onClick={[Function]}
      >
        8
      </div>
    </div>
    <div
      className="grid-item"
    >
      <div
        className="number"
        onClick={[Function]}
      >
        9
      </div>
    </div>
    <div
      className="operators"
    >
      <div
        className="grid-item"
      >
        :
      </div>
      <div
        className="grid-item"
      >
        -
      </div>
      <div
        className="grid-item"
      >
        +
      </div>
      <div
        className="grid-item"
      >
        =
      </div>
    </div>
  </div>
</div>
`;
